core/bin/external_node/src/main.rs

```diff
@@ -154,6 +158,7 @@ async fn init_tasks(
             tokio::time::sleep(Duration::from_secs(10)).await;
         }
     }));
+    let tx_cache = Arc::new(RwLock::new(TxCache::default()));
```

Comment:
Exposing `TxCache` internally looks like not the best idea; IMO, it'd make more sense to encapsulate it as with other API server tasks.